src/linesearch/armijo_wolfe.jl

```diff
@@ -1,5 +1,36 @@
 export armijo_wolfe
 
+"""
+    t, good_grad, ht, nbk, nbW = armijo_wolfe(h, h₀, slope, g)
+
+Performs a line search among the `LineModel` ``h(t) = f(x + t d)``, where
+`h₀ = h(0.0) = f(x)`, `slope = h'(0.0) = ∇f(x)ᵀd` and `g` is a which will be
+overwritten with the gradient at various points. On exit, if `good_grad=true`,
+`g` contains the gradient at the returned step length.
+The steplength is chosen to satisfy the Armijo and Wolfe conditions. The Armijo condition is
```

Comment:
In fact the steplength only attempts to satisfy the Wolfe condition. The final steplength may only satisfy Armijo's condition.